x-pack/plugins/stack_connectors/server/connector_types/sentinelone/sentinelone.ts

```diff
@@ -99,70 +127,79 @@ export class SentinelOneConnector extends SubActionConnector<
       schema: SentinelOneIsolateHostParamsSchema,
     });
 
-    this.registerSubAction({
-      name: SUB_ACTION.KILL_PROCESS,
-      method: 'killProcess',
-      schema: SentinelOneKillProcessParamsSchema,
-    });
-
     this.registerSubAction({
       name: SUB_ACTION.EXECUTE_SCRIPT,
       method: 'executeScript',
       schema: SentinelOneExecuteScriptParamsSchema,
     });
   }
 
-  public async executeScript(payload: SentinelOneExecuteScriptParams) {
-    await this.sentinelOneApiRequest({
-      url: this.urls.remoteScriptsExecute,
+  public async fetchAgentFiles({
+    files,
+    agentUUID,
+    zipPassCode,
+  }: SentinelOneFetchAgentFilesParams) {
+    const agent = await this.getAgents({ uuid: agentUUID });
+    const agentId = agent.data[0]?.id;
+
+    if (!agentId) {
+      throw new Error(`No agent found in SentinelOne for UUID [${agentUUID}]`);
+    }
+
+    return this.sentinelOneApiRequest({
+      url: `${this.urls.agents}/${agentId}/actions/fetch-files`,
       method: 'post',
       data: {
         data: {
-          outputDestination: 'SentinelCloud',
-          ...payload.script,
-        },
-        filter: {
-          computerName: payload.computerName,
+          password: zipPassCode,
+          files,
         },
       },
-      responseSchema: SentinelOneExecuteScriptResponseSchema,
+      responseSchema: SentinelOneFetchAgentFilesResponseSchema,
     });
   }
 
-  public async killProcess({ alertIds, processName, ...payload }: SentinelOneKillProcessParams) {
-    const agentData = await this.getAgents(payload);
-
-    const agentId = agentData.data[0]?.id;
+  public async downloadAgentFile({ agentUUID, activityId }: SentinelOneDownloadAgentFileParams) {
+    const agent = await this.getAgents({ uuid: agentUUID });
+    const agentId = agent.data[0]?.id;
 
     if (!agentId) {
-      throw new Error(`No agent found for filter ${JSON.stringify(payload)}`);
+      throw new Error(`No agent found in SentinelOne for UUID [${agentUUID}]`);
     }
 
-    const terminateScriptResponse = await this.getRemoteScripts({
-      query: 'terminate',
-      osTypes: agentData?.data[0]?.osType,
+    return this.sentinelOneApiRequest({
+      url: `${this.urls.agents}/${agentId}/uploads/${activityId}`,
+      method: 'get',
+      responseType: 'stream',
+      responseSchema: SentinelOneDownloadAgentFileResponseSchema,
     });
+  }
 
-    if (!processName) {
-      throw new Error('No process name provided');
+  public async getActivities(queryParams?: SentinelOneGetActivitiesParams) {
+    return this.sentinelOneApiRequest({
+      url: this.urls.activities,
+      method: 'get',
+      params: queryParams,
+      responseSchema: SentinelOneGetActivitiesResponseSchema,
+    });
+  }
+
+  public async executeScript({ filter, script }: SentinelOneExecuteScriptParams) {
```

Comment:
Should executeScript changes be included in this PR too? Just checking 👍

Reply:
yes, it was intentional. The changes here are needed in order to be able to support response actions that need to execute a script. The most important change was to ensure that this method actually returns the response from calling the S1 API